common/chains/client/rpc_client.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+)
+
+type RPCClient[
+	CHAINID types.ID,
+	SEQ types.Sequence,
+	ADDR types.Hashable,
+	BLOCK any,
```

Comment:
Haven't looked deeply yet, but i think BLOCK isn't needed. Just HEAD can be used instead of that

Reply:
Fixed!